packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -38,29 +40,31 @@ const filterableFields: (keyof EntityFilterQuery)[] = [
   'name',
   'image_url',
   'type',
-  'attributes_type',
   'generational_distance',
 ];
 const isFilterableField = (field: string): field is keyof EntityFilterQuery =>
-  (filterableFields as string[]).includes(field);
+  (filterableFields as string[]).includes(field) || field.startsWith(ATTRIBUTES_FILTER_PREFIX);
 
-const nestedFields: (keyof NestedFilterQueryFields)[] = ['attributes_type'];
 const isNestedField = (field: keyof EntityFilterQuery): field is keyof NestedFilterQueryFields =>
-  (nestedFields as (keyof EntityFilterQuery)[]).includes(field);
+  field.startsWith(ATTRIBUTES_FILTER_PREFIX);
 
 const numericFields: (keyof NumericFilterQueryFields)[] = ['generational_distance'];
 const isNumericField = (field: keyof EntityFilterQuery): field is keyof NumericFilterQueryFields =>
   (numericFields as (keyof EntityFilterQuery)[]).includes(field);
 
-type JsonBKey<
-  T extends keyof NestedFilterQueryFields
-> = T extends `${infer Field}${typeof NESTED_FIELD_DELIMITER}${infer Key}`
-  ? `${Field}${typeof JSONB_FIELD_DELIMITER}${Key}`
-  : T;
+type JsonBKey<T extends keyof NestedFilterQueryFields> =
+  T extends `${infer Field}${typeof NESTED_FIELD_DELIMITER}${infer Key}`
+    ? `${Field}${typeof JSONB_FIELD_DELIMITER}${Key}`
+    : T;
 
 const toJsonBKey = <T extends keyof NestedFilterQueryFields>(nestedField: T): JsonBKey<T> => {
-  const [field, value] = nestedField.split(NESTED_FIELD_DELIMITER);
-  return [field, value].join(JSONB_FIELD_DELIMITER) as JsonBKey<T>;
+  // get the field name, which is always the first part of the nested field
+  const field = nestedField.split(NESTED_FIELD_DELIMITER)[0];
+
+  // get the rest of the nested field, which is the nestedField without the field name and the first delimiter
+  const values = nestedField.split(`${field}_`)[1];
+
+  return [field, values].join(JSONB_FIELD_DELIMITER) as JsonBKey<T>;
```

Comment:
Ah I think the original intention was to allow for nested objects using `_`, eg.
```
filter: attributes_gym_type=fire

matches:
{
  attributes: {
    gym: {
      type: fire
    }
  }
}
```
Perhaps using `_` as the nesting delimiter was a poor choice given its so popular... I wonder if we could keep the deep comparison behaviour by switching to a different delimiter?

We'd need to change:
1. The nesting delimiter used here
2. The nesting delimiter used in the EntityApi in the api-client
3. Search in the code and database for usages of the old delimiter
- I think that would just be in the dataSourceEntityFilter in report configs